### PR TITLE
Adds a note to CONTRIBUTING.md about not adding git submodules

### PR DESCRIPTION
**Pull request by Iamgoofball**
See contents of CONTRIBUTING.md.

### CODE REVIEW

**Comment by san7890** [comment]
This is reliant on outdated information since all three of your sources were listed in 2017. Since then I have read the following:

https://stackoverflow.com/questions/1777854/how-can-i-specify-a-branch-tag-when-adding-a-git-submodule/18799234#18799234

It seems like the concerns placed in 2017 should be rejudged against the information and tooling available to us in the modern day. Aside from MSO giving a list to the issues he sees with it, I would like to know if this does not work in 2022: five years later.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -192,6 +193,14 @@ Note: VSC requires an [extension](https://marketplace.visualstudio.com/items?ite
 
 Github actions that require additional configuration are disabled on the repository until ACTION_ENABLER secret is created with non-empty value.
 
+## Can I add a Git submodule?
+### No.
+Submodules are buggy, error prone, and every time we've tried to use them, things break and it has to be removed.
+See:
+https://github.com/tgstation/tgstation/pull/28871
+https://github.com/tgstation/tgstation/pull/28902
+https://github.com/tgstation/tgstation/pull/29091
```
Comment (LemonInTheDark) [inline .github/CONTRIBUTING.md:202]: not the right place for this, also weird thing to put in the guidelines if none's adding submodules to this repo

**Comment by Mothblocks** [comment]
I agree with not wanting git submodules as a longstanding git submodule hater. We used to need them in the Roblox community because we had no good package manager for the longest time, and I would rather be dead than work with them again.

That being said, is anyone seriously considering this? All PRs you linked are from 2017. I would think anyone with both the know-how and the use case for a submodule would be in contact with maintainers anyway for us to tell them no.

**Comment by Iamgoofball** [comment]
@Mothblocks 
https://github.com/tgstation/map_depot/pull/49
https://github.com/tgstation/map_depot/pull/51

**Comment by Mothblocks** [comment]
Map depot is not a critical repository by any means (it is literally just a map archive) and the values of submodules there are clearly desired by the people who actually use it, that being the ability to more easily set up linters and MapDiffBot, without having the entirety of tgstation as part of the repository. 

As for tgstation proper, I can't imagine anyone who wants to add submodules will do so without asking maintainers and so I don't see value in adding this, especially when the cited case is on a repository that doesn't even adhere to these guidelines.

**Pull request closed without merging**